packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -47,8 +47,11 @@ export interface TerminalWidgetFactoryOptions extends Partial<TerminalWidgetOpti
 export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget {
 
     private readonly TERMINAL = 'Terminal';
+    isInteractive = true;
+    terminalId = -1;
     protected readonly onTermDidClose = new Emitter<TerminalWidget>();
-    protected terminalId = -1;
+    protected readonly onTermDidDispose = new Emitter<void>();
+    protected readonly onDidReconnect = new Emitter<TerminalWidget>();
```

Comment:
please give proper names `onDidReconnectEmitter`, `onDidReconnect` is a name for event, not an emitter